### PR TITLE
fixed get and find user position

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/CompanyService.Data/PositionRepository.cs [hunk 1]
```diff
@@ -55,6 +55,9 @@ public void Disable(Guid positionId)
 
             dbPosition.IsActive = false;
             _provider.Positions.Update(dbPosition);
+
+            dbPosition.Users.Select(u => u.IsActive = false);
```
Comment (BelehovEgor) [inline src/CompanyService.Data/PositionRepository.cs:59]: what are you doing here?

File: src/CompanyService/Controllers/PositionController.cs [hunk 2]
```diff
@@ -33,7 +33,8 @@ public class PositionController : ControllerBase
         }
 
         [HttpGet("find")]
-        public List<PositionResponse> Find([FromServices] IFindPositionsCommand command)
+        public List<PositionResponse> Find(
+            [FromServices] IFindPositionsCommand command)
```
Comment (BelehovEgor) [inline src/CompanyService/Controllers/PositionController.cs:37]: Now don't you want to create bool flag 'IncludeDisactivated'?

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/CompanyService.Broker/Consumers/ChangeUserPositionConsumer.cs [hunk 3]
```diff
@@ -14,8 +14,8 @@ public class ChangeUserPositionConsumer : IConsumer<IChangeUserPositionRequest>
 
         private object ChangePosition(IChangeUserPositionRequest request)
         {
-            _repository.Remove(request.UserId);
             bool isSuccess = _repository.Add(_mapper.Map(request.PositionId, request.UserId));
+            _repository.Remove(request.UserId);
```
Comment (TIove) [inline src/CompanyService.Broker/Consumers/ChangeUserPositionConsumer.cs:18]: for what? If you don't wanna delete position before add, then add condition for this. Now you anyway go to Remove. 
You can do following thing: if (!isSuccess) { _repository.Remove(request.UserId); }

File: src/CompanyService.Data/PositionRepository.cs (hunk 1 above)
Comment (MariyaSin) [reply src/CompanyService.Data/PositionRepository.cs]: removed

**Review by TIove — APPROVED** [review]

File: src/CompanyService.Broker/Consumers/ChangeUserPositionConsumer.cs (hunk 3 above)
Comment (MariyaSin) [reply src/CompanyService.Broker/Consumers/ChangeUserPositionConsumer.cs:18]: reworked

File: src/CompanyService/Controllers/PositionController.cs (hunk 2 above)
Comment (MariyaSin) [reply src/CompanyService/Controllers/PositionController.cs]: done

File: src/CompanyService.Business/Commands/Position/EditPositionCommand.cs [hunk 4]
```diff
@@ -44,6 +44,20 @@ public OperationResultResponse<bool> Execute(Guid positionId, JsonPatchDocument<
 
             _validator.ValidateAndThrowCustom(request);
 
+            OperationResultResponse<bool> response = new();
+
+            foreach (var item in request.Operations)
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/CompanyService.Business/Commands/Position/EditPositionCommand.cs:49]: I try to explore if i must edit IsActive. In this case, i check if the position has active positionsUsers

File: src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs [hunk 5]
```diff
@@ -21,9 +21,9 @@ public class FindPositionsCommand : IFindPositionsCommand
             _mapper = mapper;
         }
 
-        public List<PositionResponse> Execute()
+        public List<PositionResponse> Execute(bool includeDeactivated)
         {
-            return _repository.Find().Select(position => _mapper.Map(position)).ToList();
+            return _repository.Find(includeDeactivated).Select(position => _mapper.Map(position)).ToList();
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs]: done

**Review by borovHokage — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

File: src/CompanyService.Data/PositionRepository.cs [hunk 6]
```diff
@@ -39,23 +39,17 @@ public DbPosition Get(Guid? positionId, Guid? userId)
             throw new BadRequestException("You must specify 'positionId' or 'userId'.");
         }
 
-        public List<DbPosition> Find()
+        public List<DbPosition> Find(bool includeDeactivated)
         {
-            return _provider.Positions.ToList();
+            return includeDeactivated ?
+                _provider.Positions.ToList() :
+                _provider.Positions.Where(p => p.IsActive == true).ToList();
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/CompanyService.Data/PositionRepository.cs]: done

File: src/CompanyService.Data/PositionUserRepository.cs [hunk 7]
```diff
@@ -26,6 +26,11 @@ public bool Add(DbPositionUser positionUser)
                 throw new ArgumentNullException(nameof(positionUser));
             }
 
+            if (_provider.Positions.Any(p => p.Id == positionUser.PositionId && !p.IsActive))
```
Comment (TIove) [inline src/CompanyService.Data/PositionUserRepository.cs:29]: You have logic mistake in ChangeUserPositionConsumer.cs you are trying add and delete position. But if it already exists you will have this exception. That is means u will not change position if it exists

**Review by TIove — CHANGES_REQUESTED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**